src/loader.nit

```diff
@@ -746,9 +770,99 @@ redef class ModelBuilder
 				mmodule.set_visibility_for(sup, public_visibility)
 			end
 		end
-		self.toolcontext.info("{mmodule} imports {imported_modules.join(", ")}", 3)
+
+		# Declare conditional importation
+		for aimport in nmodule.n_imports do
+			if not aimport isa AStdImport then continue
+			var atconditionals = aimport.get_annotations("conditional")
+			if atconditionals.is_empty then continue
+
+			var suppath = seach_module_by_amodule_name(aimport.n_name, mmodule.mgroup)
+			if suppath == null then continue # skip error
+
+			for atconditional in atconditionals do
+				var nargs = atconditional.n_args
+				if nargs.is_empty then
+					error(atconditional, "Syntax Error: `conditional` expects module identifiers as arguments.")
+					continue
+				end
+
+				# The rule
+				var rule = new Array[Object]
+
+				# First element is the goal, thus
+				rule.add suppath
+
+				# Second element is the first condition, that is to be a client of the current module
+				rule.add mmodule
+
+				# Other condition are to be also a client of each modules indicated as arguments of the annotation
+				for narg in nargs do
+					var id = narg.as_id
+					if id == null then
+						error(narg, "Syntax Error: `conditional` expects module identifier as arguments.")
+						continue
+					end
+
+					var mp = search_mmodule_by_name(narg, mmodule.mgroup, id)
+					if mp == null then continue
+
+					rule.add mp
+				end
+
+				conditional_importations.add rule
+			end
+		end
+
 		mmodule.set_imported_mmodules(imported_modules)
 
+		# Apply conditional importations
+		# Because a conditional importation may cause additional conditional importation, use a fixed point
+		# The rules are checked naively because we assume that it does not worth to be optimized
+		var check_conditional_importations = true
+		while check_conditional_importations do
+			check_conditional_importations = false
+
+			for ci in conditional_importations do
+				# Check conditions
+				for i in [1..ci.length[ do
+					var mp = ci[i]
+					var m
+					if mp isa MModule then
+						m = mp
+					else if mp isa ModulePath then
+						m = mp.mmodule
+						# Is loaded?
+						if m == null then continue label
+					else
+						abort
+					end
+					# Is imported?
+					if not mmodule.in_importation.greaters.has(m) then continue label
+				end
+				# Still here? It means that all conditions modules are loaded and imported
+
+				# Identify the module to automatically import 
+				var suppath = ci.first.as(ModulePath)
+				var sup = load_module_path(suppath)
+				if sup == null then continue
+
+				# Do nothing if already imported
+				if mmodule.in_importation.greaters.has(sup) then continue label
+
+				# Import it
+				self.toolcontext.info("{mmodule} conditionally imports {sup}", 3)
```

Comment:
So you made a choice regarding the questions in #1579, you should document it on the issue.